app/vmui/packages/vmui/src/utils/uplot/series.ts

```diff
@@ -1,45 +1,118 @@
-import { baseContrastColors, getColorFromString } from "../color";
-import { getMedianFromArray, getMaxFromArray, getMinFromArray, getLastFromArray } from "../math";
+import { BarSeriesItem, Disp, Fill, ForecastType, HideSeriesArgs, LegendItemType, SeriesItem, Stroke } from "../../types";
+import { anomalyColors, baseContrastColors, getColorFromString } from "../color";
+import { getLastFromArray, getMaxFromArray, getMedianFromArray, getMinFromArray } from "../math";
 import { formatPrettyNumber } from "./helpers";
 
-export const getSeriesItemContext = (data: MetricResult[], hideSeries: string[], alias: string[]) => {
+// Helper function to extract freeFormFields values as a comma-separated string
+export const extractFields = (metric: MetricBase["metric"]): string => {
+  const excludeMetrics = ["__name__", "for"];
+  return Object.entries(metric)
+    .filter(([key]) => !excludeMetrics.includes(key))
+    .map(([key, value]) => `${key}: ${value}`).join(",");
+};
+
+const isForecast = (metric: MetricBase["metric"]) => {
+  const metricName = metric?.__name__ || "";
+  const forecastRegex = new RegExp(`(${Object.values(ForecastType).join("|")})$`);
+  const match = metricName.match(forecastRegex);
+  const value = match && match[0] as ForecastType;
+  return {
+    value,
+    isUpper: value === ForecastType.yhatUpper,
+    isLower: value === ForecastType.yhatLower,
+    isYhat: value === ForecastType.yhat,
+    isAnomaly: value === ForecastType.anomaly,
+    group: extractFields(metric)
+  };
+};
+
+export const getSeriesItemContext = (data: MetricResult[], hideSeries: string[], alias: string[], isAnomaly?: boolean) => {
   const colorState: {[key: string]: string} = {};
-  const stats = data.map(d => {
+  const maxColors = isAnomaly ? 0 : Math.min(data.length, baseContrastColors.length);
+
+  for (let i = 0; i < maxColors; i++) {
+    const label = getNameForMetric(data[i], alias[data[i].group - 1]);
+    colorState[label] = baseContrastColors[i];
+  }
+
+  return (d: MetricResult, i: number): SeriesItem => {
+    const forecast = isForecast(data[i].metric);
+    const label = isAnomaly ? forecast.group : getNameForMetric(d, alias[d.group - 1]);
+
     const values = d.values.map(v => promValueToNumber(v[1]));
-    return {
+    const { min, max, median, last } = {
       min: getMinFromArray(values),
       max: getMaxFromArray(values),
       median: getMedianFromArray(values),
       last: getLastFromArray(values),
     };
-  });
 
-  const maxColors = Math.min(data.length, baseContrastColors.length);
-  for (let i = 0; i < maxColors; i++) {
-    const label = getNameForMetric(data[i], alias[data[i].group - 1]);
-    colorState[label] = baseContrastColors[i];
-  }
+    let dash: number[] = [];
+    switch (true) {
+      case forecast.isLower || forecast.isUpper:
+        dash = [10, 5];
+        break;
+      case forecast.isYhat:
+        dash = [10, 2];
+        break;
+    }
```

Comment:
it is not a good idead to use switch construction in this case, because it haven't default state. it is better to use if conditions